File: app/src/main/java/com/example/chart/TradingMarkerView.java
package com.example.chart;

import android.content.Context;
import android.widget.TextView;

import com.github.mikephil.charting.components.MarkerView;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.highlight.Highlight;
import com.github.mikephil.charting.utils.MPPointF;

import java.util.List;
import java.util.Locale;

public class TradingMarkerView extends MarkerView {

    private final TextView priceText;
    private final TextView dateText;
    private List<String> dateLabels;

    public TradingMarkerView(Context context) {
        super(context, R.layout.view_chart_marker);
        priceText = findViewById(R.id.markerPrice);
        dateText  = findViewById(R.id.markerDate);
    }

    public void setDateLabels(List<String> labels) {
        this.dateLabels = labels;
    }

    @Override
    public void refreshContent(Entry e, Highlight highlight) {
        if (e != null) {
            priceText.setText("$" + String.format(Locale.US, "%.2f", e.getY()));

            int index = (int) e.getX();
            if (dateLabels != null && index >= 0 && index < dateLabels.size()) {
                dateText.setText(dateLabels.get(index));
            } else {
                dateText.setText("Index: " + index);
            }
        }
        super.refreshContent(e, highlight);
    }

    @Override
    public MPPointF getOffset() {
        return new MPPointF(-(getWidth() / 2f), -getHeight() - 20f);
    }
}

File: app/src/main/java/com/example/mogomarket/TradingMarkerView.kt
package com.example.mogomarket

import android.content.Context
import android.widget.TextView
import com.github.mikephil.charting.components.MarkerView
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.highlight.Highlight
import com.github.mikephil.charting.utils.MPPointF
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class TradingMarkerView(
    context: Context
) : MarkerView(context, R.layout.view_chart_marker) {

    private val priceText: TextView = findViewById(R.id.markerPrice)
    private val dateText: TextView = findViewById(R.id.markerDate)

    override fun refreshContent(
        e: Entry?,
        highlight: Highlight?
    ) {
        if (e != null) {
            priceText.text = "$" + String.format("%.2f", e.y)

            // אם יש Timestamp מחובר ל-Entry דרך getData()
            val timestamp = e.data
            if (timestamp != null && timestamp is Long) {
                dateText.text = SimpleDateFormat(
                    "dd MMM yyyy HH:mm",
                    Locale.getDefault()
                ).format(Date(timestamp))
            } else {
                dateText.text = "Index: ${e.x.toInt()}"
            }
        }
        super.refreshContent(e, highlight)
    }

    override fun getOffset(): MPPointF {
        return MPPointF(
            -(width / 2).toFloat(),
            -height.toFloat() - 20f
        )
    }
}